internal/consumer/cron.go

```diff
@@ -13,7 +14,12 @@ func ConfigureCron(ctx context.Context, logger *zap.Logger, config *model.Config
 	c := cron.New()
 
 	var err error
-	service, err := NewService(logger, &config.Consumer, nil)
+	databaseClient, err := database.New(ctx, logger, &config.Database)
+	if err != nil {
+		return fmt.Errorf("Unexpected error when connecting to the database. %w", err)
+	}
+	defer databaseClient.CloseConnection()
+	service, err := NewService(logger, config, nil, databaseClient)
```

Comment:
Just noticed this is passing in `nil` for the optional http client which fallbacks to creating its own version. It might be worth using the Functional Options styles to provide overrides. That way your default api will be nicer (i.e. no `nil`s) but still provides the option to swap out implementations in tests etc.

([See more](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis))

Reply:
Done, thanks for the info.